firefox-ios/Client/Frontend/Browser/Tabs/Views/TabTrayViewController.swift

```diff
@@ -531,23 +576,59 @@ final class TabTrayViewController: UIViewController,
         }
     }
 
+    private func setupBlurView() {
+        guard tabTrayUtils.isTabTrayUIExperimentsEnabled, tabTrayUtils.isTabTrayTranslucencyEnabled else { return }
+
+        // Should use Regular layout used for iPad
+        if isRegularLayout {
+            containerView.insertSubview(blurView, aboveSubview: containerView)
+
+            NSLayoutConstraint.activate([
+                blurView.topAnchor.constraint(equalTo: view.topAnchor),
+                blurView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                blurView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+                blurView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor)
+            ])
+        } else {
+            containerView.insertSubview(blurView, belowSubview: experimentSegmentControl)
+
+            NSLayoutConstraint.activate([
+                blurView.topAnchor.constraint(equalTo: experimentSegmentControl.topAnchor),
+                blurView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+                blurView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+                blurView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
+            ])
+        }
+
+        updateBlurView()
+    }
+
+    private func updateBlurView() {
+        blurView.isHidden = !tabTrayUtils.shouldBlur()
+        applyTheme()
+    }
+
     private func updateTitle() {
-        if !isTabTrayUIExperimentsEnabled, !self.isRegularLayout {
+        if !tabTrayUtils.shouldDisplayExperimentUI(), !self.isRegularLayout {
             navigationItem.title = tabTrayState.navigationTitle
         }
     }
 
     private func setupForiPad() {
         navigationItem.titleView = segmentedControl
         view.addSubviews(containerView)
+        setupBlurView()
 
         if let titleView = navigationItem.titleView {
             titleWidthConstraint = titleView.widthAnchor.constraint(equalToConstant: UX.NavigationMenu.width)
             titleWidthConstraint?.isActive = true
         }
 
+        let topConstraintTo = tabTrayUtils.isTabTrayUIExperimentsEnabled && tabTrayUtils.isTabTrayTranslucencyEnabled ?
+                                view.topAnchor : view.safeAreaLayoutGuide.topAnchor
```

Comment:
nit; easier to read?
```suggestion
        let isTabTrayEnabled = tabTrayUtils.isTabTrayUIExperimentsEnabled && tabTrayUtils.isTabTrayTranslucencyEnabled
        let topConstraintTo = isTabTrayEnabled ? view.topAnchor : view.safeAreaLayoutGuide.topAnchor
```